src/sql/platform/connection/common/connectionProfileGroup.ts

```diff
@@ -144,6 +144,7 @@ export class ConnectionProfileGroup implements IConnectionProfileGroup {
 		connections?.forEach((conn) => {
 			this._childConnections = this._childConnections.filter((curConn) => { return curConn.id !== conn.id; });
 			conn.parent = this;
+			conn.groupId = this.id;
```

Comment:
Try setting the name here and see if it doesn't break anything

Reply:
@Charles-Gagnon So it turns out that it's used in area groups such as recent connections which do not have sub server groups (and shouldn't change it), and if profiles had their name overwritten here, it would always be default even if the connection was connected from a group to start. So, assigning group name must be only in areas that need or support it, not just a one size fits all situation. (Yes, I know this is a mess, but again this is how it is so far)